exec/java-exec/src/test/java/org/apache/drill/test/TestGracefulShutdown.java

```diff
@@ -252,44 +249,40 @@ public void testRestApiShutdown() throws Exception {
     builder = enableWebServer(builder);
     QueryBuilder.QuerySummaryFuture listener;
     final String sql = "select * from dfs.root.`.`";
-    try ( ClusterFixture cluster = builder.build();
+    try (ClusterFixture cluster = builder.build();
           final ClientFixture client = cluster.clientFixture()) {
       Drillbit drillbit = cluster.drillbit("db1");
-      int port = drillbit.getContext().getConfig().getInt("drill.exec.http.port");
-      int grace_period = drillbit.getContext().getConfig().getInt(ExecConstants.GRACE_PERIOD);
+      int port = drillbit.getWebServerPort();
+      int gracePeriod = drillbit.getContext().getConfig().getInt(ExecConstants.GRACE_PERIOD);
       listener =  client.queryBuilder().sql(sql).futureSummary();
-      Thread.sleep(10000);
-
-      while( port < 8048) {
-        URL url = new URL("http://localhost:"+port+"/shutdown");
-        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-        conn.setRequestMethod("POST");
-        if (conn.getResponseCode() != 200) {
-          throw new RuntimeException("Failed : HTTP error code : "
-                  + conn.getResponseCode());
+      while (true) {
+        if (listener.isDone()) {
+          break;
         }
-        port++;
-      }
-
-      Thread.sleep(grace_period);
-
-      Collection<DrillbitEndpoint> drillbitEndpoints = cluster.drillbit()
-              .getContext()
-              .getClusterCoordinator()
-              .getAvailableEndpoints();
 
+        Thread.sleep(100L);
+      }
+      URL url = new URL("http://localhost:" + port + "/shutdown");
+      HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+      conn.setRequestMethod("POST");
+      if (conn.getResponseCode() != 200) {
+        throw new RuntimeException("Failed : HTTP error code : "
+                + conn.getResponseCode());
+      }
+      Thread.sleep(gracePeriod);
```

Comment:
Since we are already waiting for a termination condition in the loop below, I don't think we need to have this sleep here.